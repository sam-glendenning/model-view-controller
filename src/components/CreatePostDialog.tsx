import {
  Dialog,
  DialogTitle,
  DialogContent,
  Box,
  TextField,
  DialogActions,
  Button,
} from '@mui/material';
import React, { useCallback } from 'react';
import { useCreatePostController } from '@/controllers/useCreatePostController';

interface CreatePostDialogProps {
  isOpen: boolean;
  onClose: () => void;
  onSuccess?: (message: string) => void;
  onError?: (message: string) => void;
}

export const CreatePostDialog: React.FC<CreatePostDialogProps> = ({
  isOpen,
  onClose,
  onSuccess,
  onError,
}) => {
  const {
    formData,
    isSubmitDisabled,
    handleClose,
    updateTitle,
    updateBody,
    updateUserId,
    submitForm,
  } = useCreatePostController({
    onClose,
    onSuccess,
    onError,
  });

  const handleSubmit = useCallback(async () => {
    await submitForm();
  }, [submitForm]);

  return (
    <Dialog open={isOpen} onClose={handleClose} maxWidth="sm" fullWidth>
      <DialogTitle>Create New Post</DialogTitle>
      <DialogContent>
        <Box sx={{ pt: 1 }}>
          <TextField
            autoFocus
            margin="dense"
            label="Title"
            type="text"
            fullWidth
            variant="outlined"
            value={formData.title}
            onChange={e => updateTitle(e.target.value)}
            sx={{ mb: 2 }}
          />
          <TextField
            margin="dense"
            label="Content"
            type="text"
            fullWidth
            multiline
            rows={4}
            variant="outlined"
            value={formData.body}
            onChange={e => updateBody(e.target.value)}
            sx={{ mb: 2 }}
          />
          <TextField
            margin="dense"
            label="User ID"
            type="number"
            fullWidth
            variant="outlined"
            value={formData.userId}
            onChange={e => updateUserId(parseInt(e.target.value) ?? 1)}
          />
        </Box>
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose}>Cancel</Button>
        <Button
          onClick={handleSubmit}
          variant="contained"
          disabled={isSubmitDisabled}
        >
          Create
        </Button>
      </DialogActions>
    </Dialog>
  );
};
