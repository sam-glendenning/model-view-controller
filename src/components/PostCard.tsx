import React from 'react';
import {
  Card,
  CardContent,
  CardActions,
  Typography,
  Skeleton,
  Box,
  IconButton,
} from '@mui/material';
import { Edit, Delete, Visibility } from '@mui/icons-material';
import { Post } from '@/types';

interface PostCardProps {
  post?: Post;
  isLoading?: boolean;
  onView?: (post: Post) => void;
  onEdit?: (post: Post) => void;
  onDelete?: (post: Post) => void;
  showActions?: boolean;
}

export const PostCard: React.FC<PostCardProps> = ({
  post,
  isLoading = false,
  onView,
  onEdit,
  onDelete,
  showActions = true,
}) => {
  if (isLoading) {
    return (
      <Card sx={{ minHeight: 200, mb: 2 }}>
        <CardContent>
          <Skeleton variant="text" width="60%" height={32} />
          <Skeleton variant="text" width="100%" height={20} sx={{ mt: 1 }} />
          <Skeleton variant="text" width="100%" height={20} />
          <Skeleton variant="text" width="80%" height={20} />
        </CardContent>
        {showActions && (
          <CardActions>
            <Skeleton variant="rectangular" width={80} height={36} />
            <Skeleton variant="rectangular" width={80} height={36} />
            <Skeleton variant="rectangular" width={80} height={36} />
          </CardActions>
        )}
      </Card>
    );
  }

  if (!post) {
    return (
      <Card sx={{ minHeight: 200, mb: 2 }}>
        <CardContent>
          <Typography variant="h6" color="text.secondary">
            No post data available
          </Typography>
        </CardContent>
      </Card>
    );
  }

  return (
    <Card sx={{ minHeight: 200, mb: 2, transition: 'elevation 0.2s' }} elevation={2}>
      <CardContent>
        <Typography
          variant="h6"
          component="h3"
          gutterBottom
          sx={{
            overflow: 'hidden',
            textOverflow: 'ellipsis',
            display: '-webkit-box',
            WebkitLineClamp: 2,
            WebkitBoxOrient: 'vertical',
          }}
        >
          {post.title}
        </Typography>
        <Typography
          variant="body2"
          color="text.secondary"
          sx={{
            overflow: 'hidden',
            textOverflow: 'ellipsis',
            display: '-webkit-box',
            WebkitLineClamp: 3,
            WebkitBoxOrient: 'vertical',
            lineHeight: 1.4,
          }}
        >
          {post.body}
        </Typography>
        <Box sx={{ mt: 2 }}>
          <Typography variant="caption" color="text.secondary">
            Post ID: {post.id} | User ID: {post.userId}
          </Typography>
        </Box>
      </CardContent>
      {showActions && (
        <CardActions sx={{ justifyContent: 'space-between', px: 2 }}>
          <Box>
            {onView && (
              <IconButton
                size="small"
                onClick={() => onView(post)}
                aria-label="view post"
                color="primary"
              >
                <Visibility />
              </IconButton>
            )}
            {onEdit && (
              <IconButton
                size="small"
                onClick={() => onEdit(post)}
                aria-label="edit post"
                color="secondary"
              >
                <Edit />
              </IconButton>
            )}
            {onDelete && (
              <IconButton
                size="small"
                onClick={() => onDelete(post)}
                aria-label="delete post"
                color="error"
              >
                <Delete />
              </IconButton>
            )}
          </Box>
        </CardActions>
      )}
    </Card>
  );
};
