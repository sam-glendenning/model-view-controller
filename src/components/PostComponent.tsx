import React from 'react';
import {
  Card,
  CardContent,
  CardActions,
  Typography,
  Button,
  TextField,
  Box,
  Chip,
  CircularProgress,
} from '@mui/material';
import { Edit, Delete, Save, Cancel } from '@mui/icons-material';
import {
  usePostUpdateController,
  UsePostUpdateControllerProps,
} from '@/controllers/usePostUpdateController';
import {
  usePostDeleteController,
  UsePostDeleteControllerProps,
} from '@/controllers/usePostDeleteController';

interface PostComponentProps {
  post: UsePostUpdateControllerProps['post'];
  onPostDeleted?: UsePostDeleteControllerProps['onPostDeleted'];
  onPostUpdated?: UsePostUpdateControllerProps['onPostUpdated'];
  elevation?: number;
  sx?: object;
}

export const PostComponent: React.FC<PostComponentProps> = ({
  post,
  onPostDeleted,
  onPostUpdated,
  elevation = 2,
  sx = {},
}) => {
  const {
    isEditingPost,
    formData,
    isUpdateSubmitDisabled,
    startEditing,
    cancelEditing,
    updateTitle,
    updateBody,
    updateUserId,
    updatePost,
  } = usePostUpdateController({ post, onPostUpdated });

  const { isDeleteSubmitDisabled, deletePost } = usePostDeleteController({
    post,
    onPostDeleted,
  });

  const isLoading = isUpdateSubmitDisabled || isDeleteSubmitDisabled;

  return (
    <>
      <Card elevation={elevation} sx={{ mb: 2, ...sx }}>
        <CardContent>
          {isEditingPost ? (
            <Box>
              <TextField
                fullWidth
                label="Title"
                value={formData.title}
                onChange={e => updateTitle(e.target.value)}
                sx={{ mb: 2 }}
                disabled={isLoading}
              />
              <TextField
                fullWidth
                label="Content"
                multiline
                rows={4}
                value={formData.body}
                onChange={e => updateBody(e.target.value)}
                sx={{ mb: 2 }}
                disabled={isLoading}
              />
              <TextField
                fullWidth
                label="User ID"
                type="number"
                value={formData.userId}
                onChange={e => updateUserId(parseInt(e.target.value) || 1)}
                disabled={isLoading}
              />
            </Box>
          ) : (
            <Box>
              <Box
                sx={{
                  display: 'flex',
                  justifyContent: 'space-between',
                  alignItems: 'flex-start',
                  mb: 1,
                }}
              >
                <Typography
                  variant="h6"
                  component="h3"
                  sx={{ flexGrow: 1, pr: 2 }}
                >
                  {post.title}
                </Typography>
                <Chip
                  label={`Post #${post.id}`}
                  size="small"
                  color="primary"
                  variant="outlined"
                />
              </Box>

              <Typography variant="body2" color="text.secondary" sx={{ mb: 2 }}>
                {post.body}
              </Typography>

              <Chip
                label={`User ${post.userId}`}
                size="small"
                color="secondary"
                variant="outlined"
              />
            </Box>
          )}
        </CardContent>

        <CardActions sx={{ justifyContent: 'space-between', px: 2, pb: 2 }}>
          {isEditingPost ? (
            <Box sx={{ display: 'flex', gap: 1 }}>
              <Button
                startIcon={
                  isLoading ? <CircularProgress size={16} /> : <Save />
                }
                onClick={updatePost}
                variant="contained"
                color="primary"
                disabled={isUpdateSubmitDisabled || isLoading}
                size="small"
              >
                Save
              </Button>
              <Button
                startIcon={<Cancel />}
                onClick={cancelEditing}
                variant="outlined"
                disabled={isLoading}
                size="small"
              >
                Cancel
              </Button>
            </Box>
          ) : (
            <Box sx={{ display: 'flex', gap: 1 }}>
              <Button
                startIcon={<Edit />}
                onClick={startEditing}
                variant="outlined"
                color="primary"
                size="small"
                aria-label="edit post"
              >
                Edit
              </Button>
            </Box>
          )}

          <Button
            startIcon={isLoading ? <CircularProgress size={16} /> : <Delete />}
            onClick={deletePost}
            variant="outlined"
            color="error"
            disabled={isLoading}
            size="small"
            aria-label="delete post"
          >
            Delete
          </Button>
        </CardActions>
      </Card>
    </>
  );
};
