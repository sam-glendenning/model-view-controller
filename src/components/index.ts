export { PostCard } from './PostCard';
export { UserCard } from './UserCard';
export { PostsList } from './PostsList';
export { UsersList } from './UsersList';
