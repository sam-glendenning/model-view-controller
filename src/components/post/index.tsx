import React from 'react';
import { useUpdatePostController } from '@/controllers/useUpdatePostController';
import { useDeletePostController } from '@/controllers/useDeletePostController';
import { PostViewComponent } from './view/PostViewComponent';
import { PostEditComponent } from './edit/PostEditComponent';
import { DeletePostDialog } from './delete/DeletePostDialog';
import type { Post } from '@/types';

interface PostComponentProps {
  postData: Post;
  onPostDeleted?: (deletedPost: Post) => void;
  onPostUpdated?: (updatedPost: Post) => void;
}

const PostComponent: React.FC<PostComponentProps> = ({
  postData,
  onPostDeleted,
  onPostUpdated,
}) => {
  const {
    isEditingPost,
    formData,
    isUpdating,
    isSaveButtonDisabled: isUpdateSaveButtonDisabled,
    startEditing,
    cancelEditing,
    updateTitle,
    updateBody,
    updateUserId,
    updatePost,
  } = useUpdatePostController({ postData, onPostUpdated });

  const {
    isDeletePostDialogOpen,
    isDeleting,
    showDeletePostDialog,
    hideDeletePostDialog,
    confirmDelete,
  } = useDeletePostController({
    postData,
    onPostDeleted,
  });

  return (
    <div>
      {isEditingPost ? (
        <PostEditComponent
          postData={formData}
          onTitleChange={updateTitle}
          onBodyChange={updateBody}
          onUserIdChange={updateUserId}
          onSave={updatePost}
          onCancel={cancelEditing}
          onDelete={showDeletePostDialog}
          isSaveButtonDisabled={isUpdateSaveButtonDisabled}
          isUpdating={isUpdating}
          isDeleting={isDeleting}
        />
      ) : (
        <PostViewComponent
          postData={postData}
          onEditClick={startEditing}
          onDeleteClick={showDeletePostDialog}
          isDeleting={isDeleting}
        />
      )}
      <DeletePostDialog
        open={isDeletePostDialogOpen}
        postData={postData}
        confirmPostDelete={confirmDelete}
        cancelPostDelete={hideDeletePostDialog}
        isDeleting={isDeleting}
      />
    </div>
  );
};

export default PostComponent;
