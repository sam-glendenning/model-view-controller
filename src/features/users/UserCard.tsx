import React from 'react';
import {
  Avatar,
  Box,
  Card,
  CardContent,
  Skeleton,
  Typography,
} from '@mui/material';
import { Email, Language, Phone } from '@mui/icons-material';
import type { User } from '@/shared/types';

interface UserCardProps {
  user?: User;
  isLoading?: boolean;
  compact?: boolean;
}

export const UserCard: React.FC<UserCardProps> = ({
  user,
  isLoading = false,
  compact = false,
}) => {
  if (isLoading) {
    return (
      <Card sx={{ mb: 2 }}>
        <CardContent>
          <Box sx={{ display: 'flex', alignItems: 'center', mb: 2 }}>
            <Skeleton variant="circular" width={40} height={40} />
            <Box sx={{ ml: 2, flex: 1 }}>
              <Skeleton variant="text" width="60%" height={24} />
              <Skeleton variant="text" width="80%" height={20} />
            </Box>
          </Box>
          {!compact && (
            <>
              <Skeleton variant="text" width="50%" height={20} />
              <Skeleton variant="text" width="40%" height={20} />
              <Skeleton variant="text" width="60%" height={20} />
            </>
          )}
        </CardContent>
      </Card>
    );
  }

  if (!user) {
    return (
      <Card sx={{ mb: 2 }}>
        <CardContent>
          <Typography variant="h6" color="text.secondary">
            No user data available
          </Typography>
        </CardContent>
      </Card>
    );
  }

  const getInitials = (name: string) => {
    return name
      .split(' ')
      .map(n => n[0])
      .join('')
      .toUpperCase()
      .slice(0, 2);
  };

  return (
    <Card sx={{ mb: 2, transition: 'elevation 0.2s' }} elevation={1}>
      <CardContent>
        <Box
          sx={{ display: 'flex', alignItems: 'center', mb: compact ? 0 : 2 }}
        >
          <Avatar sx={{ bgcolor: 'primary.main', mr: 2 }}>
            {getInitials(user.name)}
          </Avatar>
          <Box sx={{ flex: 1 }}>
            <Typography variant="h6" component="h3">
              {user.name}
            </Typography>
            <Typography variant="body2" color="text.secondary">
              ID: {user.id}
            </Typography>
          </Box>
        </Box>

        {!compact && (
          <>
            <Box sx={{ mb: 1 }}>
              <Box sx={{ display: 'flex', alignItems: 'center', mb: 1 }}>
                <Email sx={{ fontSize: 16, mr: 1, color: 'text.secondary' }} />
                <Typography variant="body2">{user.email}</Typography>
              </Box>

              {user.phone && (
                <Box sx={{ display: 'flex', alignItems: 'center', mb: 1 }}>
                  <Phone
                    sx={{ fontSize: 16, mr: 1, color: 'text.secondary' }}
                  />
                  <Typography variant="body2">{user.phone}</Typography>
                </Box>
              )}

              {user.website && (
                <Box sx={{ display: 'flex', alignItems: 'center', mb: 1 }}>
                  <Language
                    sx={{ fontSize: 16, mr: 1, color: 'text.secondary' }}
                  />
                  <Typography variant="body2">{user.website}</Typography>
                </Box>
              )}
            </Box>

            {user.company && (
              <Box sx={{ mt: 2 }}>
                <Typography variant="subtitle2" gutterBottom>
                  Company
                </Typography>
                <Typography variant="body2" sx={{ fontWeight: 'medium' }}>
                  {user.company.name}
                </Typography>
                <Typography variant="caption" color="text.secondary">
                  {user.company.catchPhrase}
                </Typography>
              </Box>
            )}

            {user.address && (
              <Box sx={{ mt: 2 }}>
                <Typography variant="subtitle2" gutterBottom>
                  Address
                </Typography>
                <Typography variant="body2">
                  {user.address.street}, {user.address.suite}
                </Typography>
                <Typography variant="body2">
                  {user.address.city}, {user.address.zipcode}
                </Typography>
              </Box>
            )}
          </>
        )}
      </CardContent>
    </Card>
  );
};
