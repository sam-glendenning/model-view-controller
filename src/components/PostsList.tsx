import React from 'react';
import { Grid, Typography, Box, Alert, CircularProgress } from '@mui/material';
import { PostCard } from './PostCard';
import { Post } from '@/types';

interface PostsListProps {
  posts?: Post[];
  isLoading?: boolean;
  error?: string | null;
  title?: string;
  onViewPost?: (post: Post) => void;
  onEditPost?: (post: Post) => void;
  onDeletePost?: (post: Post) => void;
  showActions?: boolean;
  emptyMessage?: string;
}

export const PostsList: React.FC<PostsListProps> = ({
  posts = [],
  isLoading = false,
  error = null,
  title = 'Posts',
  onViewPost,
  onEditPost,
  onDeletePost,
  showActions = true,
  emptyMessage = 'No posts found',
}) => {
  if (error) {
    return (
      <Box sx={{ mb: 3 }}>
        <Typography variant="h5" gutterBottom>
          {title}
        </Typography>
        <Alert severity="error">
          Error loading posts: {error}
        </Alert>
      </Box>
    );
  }

  const renderLoadingSkeletons = () => {
    return Array.from({ length: 6 }, (_, index) => (
      <Grid item xs={12} sm={6} md={4} key={`skeleton-${index}`}>
        <PostCard isLoading={true} showActions={showActions} />
      </Grid>
    ));
  };

  return (
    <Box sx={{ mb: 3 }}>
      <Box sx={{ display: 'flex', alignItems: 'center', mb: 3 }}>
        <Typography variant="h5" gutterBottom sx={{ mb: 0, flexGrow: 1 }}>
          {title}
        </Typography>
        {isLoading && (
          <CircularProgress size={24} sx={{ ml: 2 }} />
        )}
      </Box>

      <Grid container spacing={2}>
        {isLoading && renderLoadingSkeletons()}
        
        {!isLoading && posts.length === 0 && (
          <Grid item xs={12}>
            <Alert severity="info">
              {emptyMessage}
            </Alert>
          </Grid>
        )}
        
        {!isLoading && posts.map((post) => (
          <Grid item xs={12} sm={6} md={4} key={post.id}>
            <PostCard
              post={post}
              onView={onViewPost}
              onEdit={onEditPost}
              onDelete={onDeletePost}
              showActions={showActions}
            />
          </Grid>
        ))}
      </Grid>
    </Box>
  );
};
