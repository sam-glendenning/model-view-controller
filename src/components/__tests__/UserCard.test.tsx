import { screen } from '@testing-library/react';
import { UserCard } from '@/components/UserCard';
import { render } from '@/test/utils';
import { mockUsers } from '@/test/mocks/handlers';

describe('UserCard', () => {
  const mockUser = mockUsers[0];

  describe('Snapshot Tests', () => {
    it('renders populated user card correctly', () => {
      const { container } = render(
        <UserCard user={mockUser} compact={false} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders compact user card correctly', () => {
      const { container } = render(
        <UserCard user={mockUser} compact={true} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders user card without optional fields correctly', () => {
      const minimalUser = {
        id: 1,
        name: 'John Doe',
        email: 'john@example.com',
      };
      
      const { container } = render(
        <UserCard user={minimalUser} compact={false} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Loading State', () => {
    it('displays loading skeleton when isLoading is true', () => {
      render(<UserCard isLoading={true} compact={false} />);
      
      const skeletons = document.querySelectorAll('.MuiSkeleton-root');
      expect(skeletons.length).toBeGreaterThan(0);
    });

    it('displays compact loading skeleton', () => {
      render(<UserCard isLoading={true} compact={true} />);
      
      const skeletons = document.querySelectorAll('.MuiSkeleton-root');
      expect(skeletons.length).toBeGreaterThan(0);
    });
  });

  describe('Error State', () => {
    it('displays no data message when user is undefined', () => {
      render(<UserCard user={undefined} />);
      
      expect(screen.getByText('No user data available')).toBeInTheDocument();
    });
  });

  describe('Content Display', () => {
    it('displays user name and ID', () => {
      render(<UserCard user={mockUser} />);
      
      expect(screen.getByText(mockUser.name)).toBeInTheDocument();
      expect(screen.getByText(`ID: ${mockUser.id}`)).toBeInTheDocument();
    });

    it('displays user initials in avatar', () => {
      render(<UserCard user={mockUser} />);
      
      const initials = mockUser.name
        .split(' ')
        .map(n => n[0])
        .join('')
        .toUpperCase()
        .slice(0, 2);
      
      expect(screen.getByText(initials)).toBeInTheDocument();
    });

    it('displays contact information in non-compact mode', () => {
      render(<UserCard user={mockUser} compact={false} />);
      
      expect(screen.getByText(mockUser.email)).toBeInTheDocument();
      if (mockUser.phone) {
        expect(screen.getByText(mockUser.phone)).toBeInTheDocument();
      }
      if (mockUser.website) {
        expect(screen.getByText(mockUser.website)).toBeInTheDocument();
      }
    });

    it('displays company information when available', () => {
      render(<UserCard user={mockUser} compact={false} />);
      
      if (mockUser.company) {
        expect(screen.getByText('Company')).toBeInTheDocument();
        expect(screen.getByText(mockUser.company.name)).toBeInTheDocument();
        expect(screen.getByText(mockUser.company.catchPhrase)).toBeInTheDocument();
      }
    });

    it('displays address information when available', () => {
      render(<UserCard user={mockUser} compact={false} />);
      
      if (mockUser.address) {
        expect(screen.getByText('Address')).toBeInTheDocument();
        expect(screen.getByText(`${mockUser.address.street}, ${mockUser.address.suite}`)).toBeInTheDocument();
        expect(screen.getByText(`${mockUser.address.city}, ${mockUser.address.zipcode}`)).toBeInTheDocument();
      }
    });

    it('hides detailed information in compact mode', () => {
      render(<UserCard user={mockUser} compact={true} />);
      
      expect(screen.queryByText('Company')).not.toBeInTheDocument();
      expect(screen.queryByText('Address')).not.toBeInTheDocument();
      if (mockUser.phone) {
        expect(screen.queryByText(mockUser.phone)).not.toBeInTheDocument();
      }
    });
  });
});
