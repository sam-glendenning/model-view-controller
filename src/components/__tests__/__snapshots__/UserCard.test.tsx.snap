// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard Snapshot Tests renders compact user card correctly 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-19ks27x-MuiPaper-root-MuiCard-root"
>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      class="MuiBox-root css-1etqjgc"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-vg3b70-MuiAvatar-root"
      >
        JD
      </div>
      <div
        class="MuiBox-root css-1rr4qq7"
      >
        <h3
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          John Doe
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          ID: 
          1
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`UserCard Snapshot Tests renders populated user card correctly 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-19ks27x-MuiPaper-root-MuiCard-root"
>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      class="MuiBox-root css-1coeexk"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-vg3b70-MuiAvatar-root"
      >
        JD
      </div>
      <div
        class="MuiBox-root css-1rr4qq7"
      >
        <h3
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          John Doe
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          ID: 
          1
        </p>
      </div>
    </div>
    <div
      class="MuiBox-root css-19idom"
    >
      <div
        class="MuiBox-root css-kmnzss"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fbi45o-MuiSvgIcon-root"
          data-testid="EmailIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 4-8 5-8-5V6l8 5 8-5z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          john.doe@example.com
        </p>
      </div>
      <div
        class="MuiBox-root css-kmnzss"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fbi45o-MuiSvgIcon-root"
          data-testid="PhoneIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          555-0123
        </p>
      </div>
      <div
        class="MuiBox-root css-kmnzss"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fbi45o-MuiSvgIcon-root"
          data-testid="LanguageIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2m6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56M12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96M4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56m2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8M12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96M14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2m.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56M16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          johndoe.com
        </p>
      </div>
    </div>
    <div
      class="MuiBox-root css-1yuhvjn"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-1oxen2w-MuiTypography-root"
      >
        Company
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body2 css-1f3fto2-MuiTypography-root"
      >
        Doe Enterprises
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
      >
        Innovation at its best
      </span>
    </div>
    <div
      class="MuiBox-root css-1yuhvjn"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-1oxen2w-MuiTypography-root"
      >
        Address
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        123 Main St
        , 
        Apt 1
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        Anytown
        , 
        12345
      </p>
    </div>
  </div>
</div>
`;

exports[`UserCard Snapshot Tests renders user card without optional fields correctly 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-19ks27x-MuiPaper-root-MuiCard-root"
>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      class="MuiBox-root css-1coeexk"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-vg3b70-MuiAvatar-root"
      >
        JD
      </div>
      <div
        class="MuiBox-root css-1rr4qq7"
      >
        <h3
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          John Doe
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          ID: 
          1
        </p>
      </div>
    </div>
    <div
      class="MuiBox-root css-19idom"
    >
      <div
        class="MuiBox-root css-kmnzss"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fbi45o-MuiSvgIcon-root"
          data-testid="EmailIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 4-8 5-8-5V6l8 5 8-5z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          john@example.com
        </p>
      </div>
    </div>
  </div>
</div>
`;
